### ISSUE TITLE
Flaky integration test - StorageState mismatch

### ISSUE THREAD

**Issue by chensun** [kubeflow/pipelines#3562]
https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/kubeflow_pipelines/3544/kubeflow-pipeline-e2e-test/1252289987068563456
```
run-api-integration-tests:	--- FAIL: TestExperimentAPI (4.30s)
run-api-integration-tests:	    --- FAIL: TestExperimentAPI/TestExperimentAPI (4.06s)
run-api-integration-tests:	        experiment_api_test.go:91: 
run-api-integration-tests:	            	Error Trace:	experiment_api_test.go:91
run-api-integration-tests:	            	Error:      	Not equal: 
run-api-integration-tests:	            	            	expected: &experiment_model.APIExperiment{CreatedAt:strfmt.DateTime{wall:0x0, ext:63723003070, loc:(*time.Location)(nil)}, Description:"my first experiment", ID:"c62d3e13-fe0e-45f5-a89c-2b66bb64c2e6", Name:"training", ResourceReferences:[]*experiment_model.APIResourceReference(nil), StorageState:"STORAGESTATE_AVAILABLE"}
run-api-integration-tests:	            	            	actual  : &experiment_model.APIExperiment{CreatedAt:strfmt.DateTime{wall:0x0, ext:63723003070, loc:(*time.Location)(nil)}, Description:"my first experiment", ID:"c62d3e13-fe0e-45f5-a89c-2b66bb64c2e6", Name:"training", ResourceReferences:[]*experiment_model.APIResourceReference(nil), StorageState:""}
run-api-integration-tests:	            	            	
run-api-integration-tests:	            	            	Diff:
run-api-integration-tests:	            	            	--- Expected
run-api-integration-tests:	            	            	+++ Actual
run-api-integration-tests:	            	            	@@ -6,3 +6,3 @@
run-api-integration-tests:	            	            	  ResourceReferences: ([]*experiment_model.APIResourceReference) <nil>,
run-api-integration-tests:	            	            	- StorageState: (experiment_model.ExperimentStorageState) (len=22) "STORAGESTATE_AVAILABLE"
run-api-integration-tests:	            	            	+ StorageState: (experiment_model.ExperimentStorageState) ""
run-api-integration-tests:	            	            	 })
run-api-integration-tests:	            	Test:       	TestExperimentAPI/TestExperimentAPI
run-api-integration-tests:	        experiment_api_test.go:120: 
run-api-integration-tests:	            	Error Trace:	experiment_api_test.go:120
run-api-integration-tests:	            	Error:      	Not equal: 
run-api-integration-tests:	            	            	expected: &experiment_model.APIExperiment{CreatedAt:strfmt.DateTime{wall:0x0, ext:63723003070, loc:(*time.Location)(nil)}, Description:"my first experiment", ID:"c62d3e13-fe0e-45f5-a89c-2b66bb64c2e6", Name:"training", ResourceReferences:[]*experiment_model.APIResourceReference(nil), StorageState:"STORAGESTATE_AVAILABLE"}
run-api-integration-tests:	            	            	actual  : &experiment_model.APIExperiment{CreatedAt:strfmt.DateTime{wall:0x0, ext:63723003070, loc:(*time.Location)(nil)}, Description:"my first experiment", ID:"c62d3e13-fe0e-45f5-a89c-2b66bb64c2e6", Name:"training", ResourceReferences:[]*experiment_model.APIResourceReference(nil), StorageState:""}
run-api-integration-tests:	            	            	
run-api-integration-tests:	            	            	Diff:
run-api-integration-tests:	            	            	--- Expected
run-api-integration-tests:	            	            	+++ Actual
run-api-integration-tests:	            	            	@@ -6,3 +6,3 @@
run-api-integration-tests:	            	            	  ResourceReferences: ([]*experiment_model.APIResourceReference) <nil>,
run-api-integration-tests:	            	            	- StorageState: (experiment_model.ExperimentStorageState) (len=22) "STORAGESTATE_AVAILABLE"
run-api-integration-tests:	            	            	+ StorageState: (experiment_model.ExperimentStorageState) ""
run-api-integration-tests:	            	            	 })
run-api-integration-tests:	            	Test:       	TestExperimentAPI/TestExperimentAPI
run-api-integration-tests:	        experiment_api_test.go:186: 
run-api-integration-tests:	            	Error Trace:	experiment_api_test.go:186
run-api-integration-tests:	            	Error:      	Not equal: 
run-api-integration-tests:	            	            	expected: &experiment_model.APIExperiment{CreatedAt:strfmt.DateTime{wall:0x0, ext:63723003070, loc:(*time.Location)(nil)}, Description:"my first experiment", ID:"c62d3e13-fe0e-45f5-a89c-2b66bb64c2e6", Name:"training", ResourceReferences:[]*experiment_model.APIResourceReference(nil), StorageState:"STORAGESTATE_AVAILABLE"}
run-api-integration-tests:	            	            	actual  : &experiment_model.APIExperiment{CreatedAt:strfmt.DateTime{wall:0x0, ext:63723003070, loc:(*time.Location)(nil)}, Description:"my first experiment", ID:"c62d3e13-fe0e-45f5-a89c-2b66bb64c2e6", Name:"training", ResourceReferences:[]*experiment_model.APIResourceReference(nil), StorageState:""}
run-api-integration-tests:	            	            	
run-api-integration-tests:	            	            	Diff:
run-api-integration-tests:	            	            	--- Expected
run-api-integration-tests:	            	            	+++ Actual
run-api-integration-tests:	            	            	@@ -6,3 +6,3 @@
run-api-integration-tests:	            	            	  ResourceReferences: ([]*experiment_model.APIResourceReference) <nil>,
run-api-integration-tests:	            	            	- StorageState: (experiment_model.ExperimentStorageState) (len=22) "STORAGESTATE_AVAILABLE"
run-api-integration-tests:	            	            	+ StorageState: (experiment_model.ExperimentStorageState) ""
run-api-integration-tests:	            	            	 })
run-api-integration-tests:	            	Test:       	TestExperimentAPI/TestExperimentAPI
run-api-integration-tests:	        experiment_api_test.go:236: 
run-api-integration-tests:	            	Error Trace:	experiment_api_test.go:236
run-api-integration-tests:	            	Error:      	Not equal: 
run-api-integration-tests:	            	            	expected: "STORAGESTATE_ARCHIVED"
run-api-integration-tests:	            	            	actual  : ""
run-api-integration-tests:	            	            	
run-api-integration-tests:	            	            	Diff:
run-api-integration-tests:	            	            	--- Expected
run-api-integration-tests:	            	            	+++ Actual
run-api-integration-tests:	            	            	@@ -1,2 +1,2 @@
run-api-integration-tests:	            	            	-(experiment_model.ExperimentStorageState) (len=21) "STORAGESTATE_ARCHIVED"
run-api-integration-tests:	            	            	+(experiment_model.ExperimentStorageState) ""
run-api-integration-tests:	            	            	 
run-api-integration-tests:	            	Test:       	TestExperimentAPI/TestExperimentAPI
run-api-integration-tests:	        experiment_api_test.go:239: 
run-api-integration-tests:	            	Error Trace:	experiment_api_test.go:239
run-api-integration-tests:	            	Error:      	Not equal: 
run-api-integration-tests:	            	            	expected: "STORAGESTATE_ARCHIVED"
run-api-integration-tests:	            	            	actual  : ""
run-api-integration-tests:	            	            	
run-api-integration-tests:	            	            	Diff:
run-api-integration-tests:	            	            	--- Expected
run-api-integration-tests:	            	            	+++ Actual
run-api-integration-tests:	            	            	@@ -1,2 +1,2 @@
run-api-integration-tests:	            	            	-(run_model.RunStorageState) (len=21) "STORAGESTATE_ARCHIVED"
run-api-integration-tests:	            	            	+(run_model.RunStorageState) ""
run-api-integration-tests:	            	            	 
run-api-integration-tests:	            	Test:       	TestExperimentAPI/TestExperimentAPI
run-api-integration-tests:	        experiment_api_test.go:242: 
run-api-integration-tests:	            	Error Trace:	experiment_api_test.go:242
run-api-integration-tests:	            	Error:      	Not equal: 
run-api-integration-tests:	            	            	expected: "STORAGESTATE_ARCHIVED"
run-api-integration-tests:	            	            	actual  : ""
run-api-integration-tests:	            	            	
run-api-integration-tests:	            	            	Diff:
run-api-integration-tests:	            	            	--- Expected
run-api-integration-tests:	            	            	+++ Actual
run-api-integration-tests:	            	            	@@ -1,2 +1,2 @@
run-api-integration-tests:	            	            	-(run_model.RunStorageState) (len=21) "STORAGESTATE_ARCHIVED"
run-api-integration-tests:	            	            	+(run_model.RunStorageState) ""
run-api-integration-tests:	            	            	 
run-api-integration-tests:	            	Test:       	TestExperimentAPI/TestExperimentAPI
run-api-integration-tests:	        experiment_api_test.go:245: 
run-api-integration-tests:	            	Error Trace:	experiment_api_test.go:245
run-api-integration-tests:	            	Error:      	Not equal: 
run-api-integration-tests:	            	            	expected: false
run-api-integration-tests:	            	            	actual  : true
run-api-integration-tests:	            	Test:       	TestExperimentAPI/TestExperimentAPI
run-api-integration-tests:	        experiment_api_test.go:248: 
run-api-integration-tests:	            	Error Trace:	experiment_api_test.go:248
run-api-integration-tests:	            	Error:      	Not equal: 
run-api-integration-tests:	            	            	expected: false
run-api-integration-tests:	            	            	actual  : true
run-api-integration-tests:	            	Test:       	TestExperimentAPI/TestExperimentAPI
run-api-integration-tests:	        experiment_api_test.go:256: 
run-api-integration-tests:	            	Error Trace:	experiment_api_test.go:256
run-api-integration-tests:	            	Error:      	Not equal: 
run-api-integration-tests:	            	            	expected: "STORAGESTATE_AVAILABLE"
run-api-integration-tests:	            	            	actual  : ""
run-api-integration-tests:	            	            	
run-api-integration-tests:	            	            	Diff:
run-api-integration-tests:	            	            	--- Expected
run-api-integration-tests:	            	            	+++ Actual
run-api-integration-tests:	            	            	@@ -1,2 +1,2 @@
run-api-integration-tests:	            	            	-(experiment_model.ExperimentStorageState) (len=22) "STORAGESTATE_AVAILABLE"
run-api-integration-tests:	            	            	+(experiment_model.ExperimentStorageState) ""
run-api-integration-tests:	            	            	 
run-api-integration-tests:	            	Test:       	TestExperimentAPI/TestExperimentAPI
run-api-integration-tests:	        experiment_api_test.go:259: 
run-api-integration-tests:	            	Error Trace:	experiment_api_test.go:259
run-api-integration-tests:	            	Error:      	Not equal: 
run-api-integration-tests:	            	            	expected: "STORAGESTATE_ARCHIVED"
run-api-integration-tests:	            	            	actual  : ""
run-api-integration-tests:	            	            	
run-api-integration-tests:	            	            	Diff:
run-api-integration-tests:	            	            	--- Expected
run-api-integration-tests:	            	            	+++ Actual
run-api-integration-tests:	            	            	@@ -1,2 +1,2 @@
run-api-integration-tests:	            	            	-(run_model.RunStorageState) (len=21) "STORAGESTATE_ARCHIVED"
run-api-integration-tests:	            	            	+(run_model.RunStorageState) ""
run-api-integration-tests:	            	            	 
run-api-integration-tests:	            	Test:       	TestExperimentAPI/TestExperimentAPI
run-api-integration-tests:	        experiment_api_test.go:262: 
run-api-integration-tests:	            	Error Trace:	experiment_api_test.go:262
run-api-integration-tests:	            	Error:      	Not equal: 
run-api-integration-tests:	            	            	expected: "STORAGESTATE_ARCHIVED"
run-api-integration-tests:	            	            	actual  : ""
run-api-integration-tests:	            	            	
run-api-integration-tests:	            	            	Diff:
run-api-integration-tests:	            	            	--- Expected
run-api-integration-tests:	            	            	+++ Actual
run-api-integration-tests:	            	            	@@ -1,2 +1,2 @@
run-api-integration-tests:	            	            	-(run_model.RunStorageState) (len=21) "STORAGESTATE_ARCHIVED"
run-api-integration-tests:	            	            	+(run_model.RunStorageState) ""
run-api-integration-tests:	            	            	 
run-api-integration-tests:	            	Test:       	TestExperimentAPI/TestExperimentAPI
run-api-integration-tests:	        experiment_api_test.go:265: 
run-api-integration-tests:	            	Error Trace:	experiment_api_test.go:265
run-api-integration-tests:	            	Error:      	Not equal: 
run-api-integration-tests:	            	            	expected: false
run-api-integration-tests:	            	            	actual  : true
run-api-integration-tests:	            	Test:       	TestExperimentAPI/TestExperimentAPI
run-api-integration-tests:	        experiment_api_test.go:268: 
run-api-integration-tests:	            	Error Trace:	experiment_api_test.go:268
run-api-integration-tests:	            	Error:      	Not equal: 
run-api-integration-tests:	            	            	expected: false
run-api-integration-tests:	            	            	actual  : true
run-api-integration-tests:	            	Test:       	TestExperimentAPI/TestExperimentAPI
```
/cc @jingzhang36 @hongye-sun

**Comment by chensun** [comment]
Somehow we're not using the latest build of api-server. 
https://screenshot.googleplex.com/mNg7LcvPBx7

I can also confirm this as the added log not showing up:
https://screenshot.googleplex.com/HX6pwQbFEHD
https://screenshot.googleplex.com/OH5G8nX4zBK

**Comment by chensun** [comment]
This was fixed by #3572
